examples/src/bin/texture.rs

```diff
@@ -688,149 +641,97 @@ fn main() {
             .color_blend_state(&color_blend_state)
             .dynamic_state(&dynamic_state_info)
             .layout(pipeline_layout)
-            .render_pass(renderpass);
+            .render_pass(base.renderpass);
 
         let graphics_pipelines = base
             .device
             .create_graphics_pipelines(
                 vk::PipelineCache::null(),
-                &[graphic_pipeline_infos.build()],
+                &[graphics_pipeline_info.build()],
                 None,
             )
             .unwrap();
 
-        let graphic_pipeline = graphics_pipelines[0];
-
-        base.render_loop(|| {
-            let (present_index, _) = base
-                .swapchain_loader
-                .acquire_next_image(
-                    base.swapchain,
-                    std::u64::MAX,
-                    base.present_complete_semaphore,
-                    vk::Fence::null(),
-                )
-                .unwrap();
-            let clear_values = [
-                vk::ClearValue {
-                    color: vk::ClearColorValue {
-                        float32: [0.0, 0.0, 0.0, 0.0],
+        let graphics_pipeline = graphics_pipelines[0];
+
+        let mut can_render = true;
```

Comment:
Not really sure why this variable is needed to be honest but without it the program segfaults when you close the window. The problem is after it goes into CloseRequested the event loop loops back round again and goes into RedrawEventsCleared where it calls the render functions with already destroyed variables hence the segfault. I'm not very experienced with winit at all but I would have assumed after CloseRequested it would just break?

Reply:
I think you need to render on the `RedrawRequested` event instead of the `RedrawEventsCleared`

Reply:
In fact, you should render in `MainEventsCleared`. [See the winit docs.](https://docs.rs/winit/0.24.0/winit/event/enum.Event.html#variant.MainEventsCleared)

Reply:
> I would have assumed after CloseRequested it would just break

You should probably put the cleanup code in [`LoopDestroyed`](https://docs.rs/winit/0.24.0/winit/event/enum.Event.html#variant.LoopDestroyed).

Reply:
Ya this solved it :+1: